server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSMailQueue.java

```diff
@@ -429,14 +436,15 @@ protected void populateMail(Message message, MailImpl mail) throws JMSException
      * @param message The attribute source.
      * @param mail    The mail on which attribute should be set.
      * @param name    The attribute name.
+     * @throws IOException 
      */
-    private void setMailAttribute(Message message, Mail mail, String name) {
+    private void setMailAttribute(Message message, Mail mail, String name) throws IOException {
         // Now cast the property back to Serializable and set it as attribute.
         // See JAMES-1241
         Object attrValue = Throwing.function(message::getObjectProperty).apply(name);
 
         if (attrValue instanceof String) {
-            mail.setAttribute(name, SerializationUtil.deserialize((String) attrValue));
+            mail.setAttribute(new Attribute(AttributeName.of(name), AttributeValue.fromJsonString((String) attrValue)));
```

Comment:
Please do document in unreleased that one need to do the upgrade **with an empty JMS queue**.

To empty the JMS queue, start a James with SMTP, LMTP & JMAP being disabled, and wait for the queue to be empty.